### ISSUE TITLE
Incorrect sensor State / Due value occasionally being returned

### ISSUE THREAD

**Issue by CarlBird** [BottlecapDave/HomeAssistant-FirstBus#11]
### Description
Thanks for developing this tool. I have been meaning to make use of it for a long time now and have finally got round to doing so. I have been attempting to troubleshoot an issue that I thought was in my own code, but it appears that it is the sensor state that is causing the issue.

In short, the issue is that the sensor state and is occasionally returning an incorrect value (-1), presumably due to the 'Due'. time being processing incorrectly.

**Example::**

_State correctly shown (Due at 09:11):_
State: 18
ServiceRef: '0'
ServiceNumber: '99'
Destination: NEWCASTLE, Bus Stn
Due: '2023-07-04T09:11:00+01:00'
IsFG: 'N'
IsLive: 'N'
stop: 3800C503101

_The API response made a few minutes after the above (Due in 16 mins):_
[](https://www.firstbus.co.uk/api/get-next-bus?stop=3800C503101)
`{"id":"http:\/\/first.transportapi.com\/v3\/uk\/bus\/stop_timetables\/3800C503101.json?app_id=75da7e19\u0026app_key=8a756a1369a55907a33042def6008ec8\u0026group=false\u0026limit=20\u0026live=true","atcocode":"3800C503101","smscode":"stadtdag","request_time":"2023-07-04T08:59:42+01:00","name":"Marsh Avenue","stop_name":"Marsh Avenue","bearing":"SE","indicator":"opp","locality":"Wolstanton, Newcastle Under Lyme","location":{"type":"Point","coordinates":[-2.22027,53.03381]},"departures":{},"times":[**{"ServiceRef":"0","ServiceNumber":"99","Destination":"NEWCASTLE, Bus Stn","Due":"16 mins","IsFG":"N","IsLive":"Y"}**,{"ServiceRef":"0","ServiceNumber":"98","Destination":"NEWCASTLE, Bus Stn","Due":"19 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"99","Destination":"NEWCASTLE, Bus Stn","Due":"33 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"98","Destination":"NEWCASTLE, Bus Stn","Due":"49 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"94B","Destination":"Newcastle Town Centr","Due":"09:55","IsFG":"N","IsLive":"N"},{"ServiceRef":"0","ServiceNumber":"99","Destination":"NEWCASTLE, Bus Stn","Due":"66 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"98","Destination":"NEWCASTLE, Bus Stn","Due":"87 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"99","Destination":"NEWCASTLE, Bus Stn","Due":"93 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"98","Destination":"NEWCASTLE, Bus Stn","Due":"109 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"94B","Destination":"Newcastle Town Centr","Due":"10:55","IsFG":"N","IsLive":"N"}],"stop":{"code":"3800C503101","mark":"1","sms":"stadtdag","direction":"SE","latitude":"53.033830000","longitude":"-2.220270000","indicator":"Opp","name":"stop Opp Marsh Avenue","locality":"Wolstanton","qualifier":"Newcastle Under Lyme"},"source":"(n\/a)","fromTraveline":"04\/07\/2023 08:59"}`

_The sensor state then changes to '-1' a few minutes later (note that Due has changed to 09:01):_
State: -1
ServiceRef: '0'
ServiceNumber: '99'
Destination: Newcastle
Due: '2023-07-04T09:01:00+01:00'
IsFG: 'N'
IsLive: 'N'
stop: 3800C503101

_The API response from the above (note that Due correctly shows 14 minutes):_
[](https://www.firstbus.co.uk/api/get-next-bus?stop=3800C503101)
`{"id":"http:\/\/first.transportapi.com\/v3\/uk\/bus\/stop_timetables\/3800C503101.json?app_id=75da7e19\u0026app_key=8a756a1369a55907a33042def6008ec8\u0026group=false\u0026limit=20\u0026live=true","atcocode":"3800C503101","smscode":"stadtdag","request_time":"2023-07-04T09:01:12+01:00","name":"Marsh Avenue","stop_name":"Marsh Avenue","bearing":"SE","indicator":"opp","locality":"Wolstanton, Newcastle Under Lyme","location":{"type":"Point","coordinates":[-2.22027,53.03381]},"departures":{},"times":[{"ServiceRef":"0","ServiceNumber":"99","Destination":"NEWCASTLE, Bus Stn","Due":"14 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"98","Destination":"NEWCASTLE, Bus Stn","Due":"17 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"99","Destination":"NEWCASTLE, Bus Stn","Due":"31 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"98","Destination":"NEWCASTLE, Bus Stn","Due":"47 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"94B","Destination":"Newcastle Town Centr","Due":"09:55","IsFG":"N","IsLive":"N"},{"ServiceRef":"0","ServiceNumber":"99","Destination":"NEWCASTLE, Bus Stn","Due":"64 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"98","Destination":"NEWCASTLE, Bus Stn","Due":"85 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"99","Destination":"NEWCASTLE, Bus Stn","Due":"91 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"98","Destination":"NEWCASTLE, Bus Stn","Due":"107 mins","IsFG":"N","IsLive":"Y"},{"ServiceRef":"0","ServiceNumber":"94B","Destination":"Newcastle Town Centr","Due":"10:55","IsFG":"N","IsLive":"N"}],"stop":{"code":"3800C503101","mark":"1","sms":"stadtdag","direction":"SE","latitude":"53.033830000","longitude":"-2.220270000","indicator":"Opp","name":"stop Opp Marsh Avenue","locality":"Wolstanton","qualifier":"Newcastle Under Lyme"},"source":"(n\/a)","fromTraveline":"04\/07\/2023 09:01"}`

### To Reproduce
Monitor the above API call and observe that

### Expected behavior
The correct sensor State / Due value's should be returned.

### Home Assistant Logs
N/A

**Comment by BottlecapDave** [comment]
Hi there. The state can go to negative numbers for a number of reasons. 1) there are no buses scheduled. I'm wanting to change this to report none, but this would be a breaking change so I've been putting it off or 2) the bus is overdue.

The integration only updates the data every 5 minutes so that the API isn't hammered and try and go under the radar as it's not an official API. This means there is a 5 minute period where the integration works out the remaining minutes comparing now with the last known time.

I believe in your scenario the data has not refreshed and so the sensor has gone into an "overdue" state, which is expected.

**Comment by BottlecapDave** [comment]
Sorry for the delayed update.

There was a bug in the sensor where it would show -1 minutes if the bus was only overdue by seconds. This has been fixed in the latest release (https://github.com/BottlecapDave/HomeAssistant-FirstBus/releases/tag/v1.1.3)

Again based on the frequency the sensor pulls down information, there's a chance it could still go negative, but this is reduced. I've also updated the docs based on what was said here and to state if there is no known next bus, that the sensor could go to `none`/`unknown` (existing behaviour).
